jax/_src/numpy/ufunc_api.py

```diff
@@ -32,6 +33,37 @@
 import numpy as np
 
 
+def get_if_single_primitive(fun, *args):
+  """
+  If fun(*args) lowers to a single primitive with inputs and outputs matching function
+  inputs and outputs, return that primitive. Otherwise return None.
+  """
+  try:
+    jaxpr = jax.make_jaxpr(fun)(*args)
+  except:
+    return None
+  while len(jaxpr.eqns) == 1:
+    if (jaxpr.eqns[0].invars, jaxpr.eqns[0].outvars) != (jaxpr.jaxpr.invars, jaxpr.jaxpr.outvars):
+      return None
+    elif jaxpr.eqns[0].primitive == jax._src.pjit.pjit_p:
```

Comment:
Maybe we should check that the `pjit_p` has trivial parameters like `in_shardings` / `out_shardings` (and possibly others) that can safely be ignored here (since we effectively drop them when we recurse into the jaxpr). Those seem like edge cases but it might be odd to silently ignore some of those.